### ISSUE TITLE
[BITV] Header bar a11y issues

### ISSUE THREAD

**Issue by PVince81** [nextcloud/server#33741]
Need to recheck those for #33728 @juliushaertl @CarlSchwan 

* [ ] #35201
* [x] Header navigation needs to be a `<nav>` landmark
* [x] Logo link should not be h1, but rather have aria-label="to Nextcloud landing page"
* [x] #35138
* [x] active app needs to be tagged as aria-current="page"
* [x] "More" menu icon needs to be a `<button>`, not a link
* [x] When the menu is not focused anymore, it should be closed
* [x] On mobile, active app needs better indication (e.g. via a dot)

**Comment by juliushaertl** [comment]
Checked everything that is addressed in the above PR.

> Keyboard focus overlaps with text

I've been talking with @jancborchardt about this already, but we don't have a good idea how to solve that nicely yet.

> Logo link should not be h1, but rather have aria-label="to Nextcloud landing page"
>  h1 should instead be in the app content, ideally displayed visually

I'll keep this separate from the app menu pr to get it in more quickly for the UI redesign.

app content however might not be suitable as it is replaced by any vue app, if we need it in the content, we need some concept how that would look for the individual apps first @jancborchardt

**Comment by JuliaKirschenheuter** [comment]
> * Logo link should not be h1, but rather have aria-label="to Nextcloud landing page"

> Logo link should not be h1
Already done in https://github.com/nextcloud/server/pull/35071

> h1 should instead be in the app content, ideally displayed visually
I'll move it to content. But in our design it is not easy to place <h1> visually somewhere in a content. For now i will leave it only for screen readers (invisible) in a content.

**Comment by JuliaKirschenheuter** [comment]
> * Keyboard focus overlaps with text

@PVince81 you mean 

![Screenshot from 2022-11-15 11-27-55](https://user-images.githubusercontent.com/6078378/201896562-bea02efc-132d-4081-8b39-a87b8c58d0a2.png) ?

Then i guess we need another design
